app/assets/src/components/common/CoverageVizBottomSidebar/CoverageVizBottomSidebar.jsx

```diff
@@ -137,21 +145,47 @@ export default class CoverageVizBottomSidebar extends React.Component {
     this.loadAccession(accession);
   };
 
-  generateCoverageVizData = (coverageData, coverageBinSize) =>
-    coverageData.map((value, index) => ({
-      x0: index * coverageBinSize,
-      length: value
-    }));
+  handleHistogramHover = (pageX, pageY, hoverData) => {
+    const { currentAccessionData } = this.state;
+
+    if (hoverData && hoverData[0] === 0) {
+      this.setState({
+        histogramTooltipParams: {
+          left: pageX,
+          top: pageY,
+          tooltipData: getHistogramTooltipData(
```

Comment:
I think there is a better way to do this... I would have to understand the context better.
But you typically set state fo the hovered element, and get the data on re-rendering if it changes. 
Definitely, left and top should not be in state.

Reply:
Refactored into separate "enter", "hover", "exit" events as discussed.